File: Protocolo-Colaborativo/arreglos/02_recorrido_arreglo.js
/**
 * Protocolo Colaborativo - Unidad 1: Estructuras de Datos
 * Actividad 1.2: Recorrido e impresión mediante bucle clásico y for-each.
 * Lenguaje: JavaScript (Node.js)
 */

function main() {
    const valores = [15, 28, 33, 42, 59];

    console.log("=== Actividad 2: Recorridos de Arreglo (JavaScript) ===");

    // 1. Recorrido con bucle for clásico (indexado tradicional)
    console.log("\n1. Recorrido clásico indexado (for i = 0):");
    for (let i = 0; i < valores.length; i++) {
        console.log(`Índice ${i} -> Valor: ${valores[i]}`);
    }

    // 2. Recorrido con for-each (iteración directa con for...of)
    console.log("\n2. Recorrido for-each (for...of):");
    for (const elemento of valores) {
        console.log(`Elemento: ${elemento}`);
    }
}

main();

File: Protocolo-Colaborativo/arreglos/02_recorrido_arreglo.py
"""
Protocolo Colaborativo - Unidad 1: Estructuras de Datos
Actividad 1.2: Recorrido e impresión mediante bucle clásico y for-each.
Lenguaje: Python 3
"""

def main():
    valores = [15, 28, 33, 42, 59]

    print("=== Actividad 2: Recorridos de Arreglo (Python) ===")
    
    # 1. Recorrido con bucle for clásico (indexado por rango)
    print("\n1. Recorrido clásico indexado (for i in range):")
    for i in range(len(valores)):
        print(f"Índice {i} -> Valor: {valores[i]}")

    # 2. Recorrido con for-each (iteración directa sobre elementos)
    print("\n2. Recorrido for-each (for elemento in valores):")
    for elemento in valores:
        print(f"Elemento: {elemento}")

if __name__ == "__main__":
    main()
